cloudinit/cmd/main.py

```diff
@@ -764,6 +765,24 @@ def status_wrapper(name, args):
 
     v1 = status["v1"]
     v1["stage"] = mode
+    start_time = v1[mode]["start"]
```

Comment:
I think we should drop most of this section. It gives us a bunch of recoverable errors, and it winds up giving a messy status (that's also shown twice?)
```
root@me:~# cloud-init status --long
status: error
extended_status: error - done
boot_status_code: enabled-by-generator
last_update: Thu, 25 Apr 2024 20:00:25 +0000
detail: DataSourceLXD
errors:
	- sys.exit(1) called
recoverable_errors:
WARNING:
	- [modules-final] Unexpected value found in status.json. Was this stage restarted?
	- unexpected start_time: 1714075207.5654504
	- unexpected errors: ['sys.exit(1) called']
	- unexpected recoverable_errors: {'ERROR': ['Cloud-init 24.1.3 received SIGTERM, exiting...\n  Filename: /usr/lib/python3.12/subprocess.py\n  Function: _try_wait\n  Line number: 2011\n    Filename: /usr/lib/python3.12/subprocess.py\n    Function: _wait\n    Line number: 2053\n      Filename: /usr/lib/python3.12/subprocess.py\n      Function: wait\n      Line number: 1264', 'failed stage modules-final']}
	- [modules-final] Unexpected value found in status.json. Was this stage restarted?
	- unexpected start_time: 1714075207.5654504
	- unexpected errors: ['sys.exit(1) called']
	- unexpected recoverable_errors: {'ERROR': ['Cloud-init 24.1.3 received SIGTERM, exiting...\n  Filename: /usr/lib/python3.12/subprocess.py\n  Function: _try_wait\n  Line number: 2011\n    Filename: /usr/lib/python3.12/subprocess.py\n    Function: _wait\n    Line number: 2053\n      Filename: /usr/lib/python3.12/subprocess.py\n      Function: wait\n      Line number: 1264', 'failed stage modules-final']}
```

Way further down there's a section like this:
```python
        # merge new recoverable errors into exising recoverable error list
        old_recoverable_errors = v1[mode]["recoverable_errors"]
        new_recoverable_errors = next(
            filter(
                lambda h: isinstance(h, log.LogExporter), root_logger.handlers
            )
        ).export_logs()
        for key in new_recoverable_errors.keys():
            if key in old_recoverable_errors:
                old_recoverable_errors[key].extend(new_recoverable_errors[key])
            else:
                old_recoverable_errors[key] = new_recoverable_errors[key]
```

Could we turn this section into it's own function, and then rather than
```python
    v1[mode]["recoverable_errors"] = next(
        filter(lambda h: isinstance(h, log.LogExporter), root_logger.handlers)
    ).export_logs()
```

we call the new function to add the new recoverable errors to the old ones? The first run of cloud-final will log that it got a sigterm. I would think that should give enough signal as to what happened rather than introducing new warnings that are less precise.

Reply:
Oops, those duplicates definitely aren't desired.

> The first run of cloud-final will log that it got a sigterm

That covers systemd's default behavior, yes, but what about if a SIGSTOP or SIGKILL happens?

If cloud-init receives a SIGSTOP or SIGKILL, our logging and `cloud-init status` tooling will not know that this happened. This section was a proposed solution to that problem.

Reply:
That makes sense. So here, can we check that start time exists, but finished time does not (indicating a run that started but never finished), and then combine the errors/recoverable errors to the output of the next run?